### PR TITLE
Feat/bac 12 transcription service

### PR DESCRIPTION
**Pull request by Tamunonengiyeofori**
I have created the transcripts.py file and updated it with endpoints to retrieve a transcript by ID get all transcripts

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 # Local Netlify folder
```
Comment (ColeDrain) [inline .gitignore:1]: where is this coming from, kindly delete ?

File: backend/app/routers/transcript.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+@transcript_router.get("/{job_id}", description="Retrieving transcript by audio ID")
+def view_transcript(job_id: Union[int, str], db: Session = Depends(_services.get_session)):
+    Job = db.query(models.Job).filter(models.Job.id == job_id).first()
+    job_audio_id = Job.audio_id
+    transcript_audio = db.query(models.Audio).filter(models.Audio.id == job_audio_id).first()
+    transcript_text = transcript_audio.transcript
+    return transcript_text
+
+
+#ENDPOINT TO GET ALL TRANSCRIPTS AS A LIST
```
Comment (ColeDrain) [inline backend/app/routers/transcript.py:30]: This brings integrity issues, you are getting transcripts of everyone, I don't think this is something that should happen.
Transcripts are personal to companies/users.

**Review by ColeDrain — CHANGES_REQUESTED** [review]
Good one, but some issues that would break our app.

- requirements.txt file is outside of backend/app directory and contains unneeded 
dependencies, please resolve

- Delete routes/view_transcripts.py instead of commenting out

- Please fix your directory structure, it doesn't follow dev or main, some folders are outside backend/app (our working directory)

- Why do we have trascript.py again, put the functionality into transcribe.py

- .gitignore file outside of working directory

- And other comments I made

File: backend/app/routers/transcript.py (hunk 2 above)
Comment (Tamunonengiyeofori) [reply backend/app/routers/transcript.py:30]: I was told by the frontend dev to get the transcripts of everyone

**Comment by Tamunonengiyeofori** [comment]
I have resolved the merge conflicts

File: backend/app/main.py [hunk 3]
```diff
@@ -219,6 +220,9 @@ def update_user(user: schema.user_update, user_id: int, db:Session=_fastapi.Depe
      return crud.update_user(db=db, user=user, user_id=user_id)
 
 
+
+@app.get("/new_analysis/{id}", response_model=schema.Analysis, tags=['analysis'])
+=======
```
Comment (ColeDrain) [inline backend/app/main.py:225]: Bro please fix this (=====)

File: backend/app/main.py [hunk 4]
```diff
@@ -231,6 +235,7 @@ async def get_history(user: models.User = Depends(get_current_user), db: Session
 
 
 @app.get("/new_analysis/{id}", summary = "get result of a sentiment analysis", response_model=schema.Analysis, tags=['analysis'])
+
```
Comment (ColeDrain) [inline backend/app/main.py:238]: please also remove this line please, for the sake of clarity

File: backend/app/routers/transcribe.py [hunk 5]
```diff
@@ -1,14 +1,44 @@
...
+import base64
+import banana_dev as banana
+
 from . import utility as utils
 from dotenv import load_dotenv
 import os
 
 load_dotenv()
 
+def transcribe_file(filename):
```
Comment (ColeDrain) [inline backend/app/routers/transcribe.py:19]: did you sync with @DavidIkechi 
we are using Assembly AI, so Banana dev code should be removed

File: backend/app/routers/transcribe.py [hunk 6]
```diff
@@ -34,6 +64,29 @@ def get_transcript(filename):
...
+        
+    return new_paragraph
+
+        
+# """ Please Note that these endpoints are subject to change as the query would be better suited to retrieve transcripts from the transcript table by transcript_id and
+#     not Audio by audio_id.
+#     If the transcript table is available, the code will be refractored to implement changes and queries to the transcript table """
+
+# ENDPOINT TO GET A PARTICULAR TRANSCRIPT USING THE AUDIO ID
+@transcript_router.get("/{job_id}", description="Retrieving transcript by audio ID")
```
Comment (ColeDrain) [inline backend/app/routers/transcribe.py:76]: please protect this endpoint
only logged in user should have access to this

File: backend/app/routers/transcribe.py [hunk 7]
```diff
@@ -34,6 +64,29 @@ def get_transcript(filename):
...
+def view_transcript(job_id: Union[int, str], db: Session = Depends(_services.get_session)):
+    Job = db.query(models.Job).filter(models.Job.id == job_id).first()
+    job_audio_id = Job.audio_id
+    transcript_audio = db.query(models.Audio).filter(models.Audio.id == job_audio_id).first()
+    transcript_text = transcript_audio.transcript
+    return transcript_text
+
+
+#ENDPOINT TO GET ALL TRANSCRIPTS AS A LIST
+@transcript_router.get("/view_transcripts", response_model=List[schema.Audio], description="Retrieve all Transcripts")
```
Comment (ColeDrain) [inline backend/app/routers/transcribe.py:86]: same as above here

**Review by ColeDrain — CHANGES_REQUESTED** [review]

File: backend/app/routers/transcribe.py (hunk 5 above)
Comment (Tamunonengiyeofori) [reply backend/app/routers/transcribe.py:19]: I have spoken with him and corrected the code for that.

**Comment by DavidIkechi** [comment]
please I need to test your code to ensure it's working before it can be merged

**Comment by DavidIkechi** [comment]
If you are ready, we can go on a call, to avoid any break on the server.

File: backend/app/routers/transcribe.py (hunk 6 above)
Comment (Tamunonengiyeofori) [reply backend/app/routers/transcribe.py:76]: I have resolved this issue

**Review by DavidIkechi — APPROVED** [review]
Weldone bro, this looks better

**Review by ColeDrain — APPROVED** [review]

**Pull request merged**